lib/runner.js

```diff
@@ -121,18 +121,30 @@ module.exports = Runner;
  * @extends external:EventEmitter
  * @public
  * @class
- * @param {Suite} suite Root suite
- * @param {boolean} [delay] Whether to delay execution of root suite until ready.
+ * @param {Suite} suite - Root suite
+ * @param {Object|boolean} [opts] - Options. If `boolean`, whether or not to delay execution of root suite until ready (for backwards compatibility).
+ * @param {boolean} [opts.delay] - Whether to delay execution of root suite until ready.
+ * @param {boolean} [opts.cleanReferencesAfterRun] - Whether to clean references to test fns and hooks when a suite is done.
  */
-function Runner(suite, delay) {
+function Runner(suite, opts) {
+  if (opts === undefined) {
+    opts = {};
+  }
+  if (typeof opts === 'boolean') {
```

Comment:
It's good that this change is backwards compatible. But if the updated pattern is something we'd like to go ahead with in the future, maybe we should consider a deprecation notice here and queue a removal in a future major release

Reply:
maybe, but we could also just keep it in, since it doesn't look too hairy (at least for now!).  IMO, if/when it becomes a maintenance burden I'd advocate for this approach.